drivers/platform/aducm3029/pwm.c

```diff
@@ -0,0 +1,300 @@
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include "pwm.h"
+#include "stdlib.h"
+#include "error.h"
+#include <drivers/tmr/adi_tmr.h>
+#include <drivers/pwr/adi_pwr.h>
+
+#define NS_PER_SEC 1000000000
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+static int32_t config_clock_init(ADI_TMR_CONFIG *cfg, uint32_t period,
+				 uint32_t duty, uint32_t *match)
+{
+	float		base_time;
+	float		aux;
+	uint32_t	clk_freq;
+	int32_t		ret;
+	/*
+	 * ADI_TMR_CLOCK_HFOSC (26MHz) &&  ADI_TMR_PRESCALER_1
+	 * -> base_period = 38nS -> max_period = 2,490330 ms
+	 * ADI_TMR_CLOCK_HFOSC (26MHz) &&  ADI_TMR_PRESCALER_16
+	 * -> base_period = 615nS -> max_period = 40,304025 ms
+	 * ADI_TMR_CLOCK_HFOSC (26MHz) &&  ADI_TMR_PRESCALER_64
+	 * -> base_period = 2461nS -> max_period = 161,281635 ms
+	 * ADI_TMR_CLOCK_HFOSC (26MHz) &&  ADI_TMR_PRESCALER_256
+	 * -> base_period = 9846nS -> max_period = 645,257610 ms
+	 *
+	 * ADI_TMR_CLOCK_LFOSC (32KHz) &&  ADI_TMR_PRESCALER_1
+	 * -> base_period = 31,25uS -> max_period = 2,04796875 s
+	 * ADI_TMR_CLOCK_LFOSC (32KHz) &&  ADI_TMR_PRESCALER_16
+	 * -> base_period = 500uS -> max_period = 32,767500 s
+	 * ADI_TMR_CLOCK_LFOSC (32KHz) &&  ADI_TMR_PRESCALER_64
+	 * -> base_period = 2mS -> max_period = 131,070 s
+	 * ADI_TMR_CLOCK_LFOSC (32KHz) &&  ADI_TMR_PRESCALER_256
+	 * -> base_period = 8mS -> max_period = 524,280 s
+	 */
+
+	/* Use HFOSC because in general frequency
+	 * for pwms are lower than 645 ms */
```

Comment:
nitpick: a bit confusing talking about a frequency being lower than a period of time